partition_table.go

```diff
@@ -424,43 +404,27 @@ func (p *PartitionTable) markRecovered(ctx context.Context) error {
 	}
 }
 
-func (p *PartitionTable) handleConsumerErrors(ctx context.Context, errs *multierr.Errors, cons sarama.PartitionConsumer) {
-	for {
-		select {
-		case consError, ok := <-cons.Errors():
-			if !ok {
-				return
-			}
-			err := fmt.Errorf("Consumer error: %v", consError)
-			p.log.Printf("%v", err)
-			errs.Collect(err)
-			// if there's an error, close the consumer
-			cons.AsyncClose()
-		case <-ctx.Done():
-			return
-		}
-	}
-}
+func (p *PartitionTable) drainConsumer(cons sarama.PartitionConsumer) error {
 
-func (p *PartitionTable) drainConsumer(cons sarama.PartitionConsumer, errs *multierr.Errors) {
-
-	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+	timeoutCtx, cancel := context.WithTimeout(context.Background(), consumerDrainTimeout)
 	defer cancel()
 
-	errg, ctx := multierr.NewErrGroup(ctx)
+	errg, _ := multierr.NewErrGroup(context.Background())
 
 	// drain errors channel
 	errg.Go(func() error {
+		var errs *multierror.Error
+
 		for {
 			select {
-			case <-ctx.Done():
+			case <-timeoutCtx.Done():
 				p.log.Printf("draining errors channel timed out")
-				return nil
+				return errs
```

Comment:
shouldn't return here `nil`?

Reply:
i thought we should still return the errors we have collected so far, even if it times out.